evm/gas-service.js

```diff
@@ -0,0 +1,211 @@
+                return null;
+            }
+
+            const { gasEstimationType = 0, blobBaseFee = 0 } = destinationConfig;
+
+            const data = await httpPost(`${api}/gmp/getFees`, {
+                sourceChain: chain.axelarId,
+                destinationChain,
+                sourceTokenAddress: '0x0000000000000000000000000000000000000000',
+            });
+
+            const {
+                source_base_fee: sourceBaseFee,
+                source_token: {
+                    gas_price_in_units: { value: gasPrice },
+                    token_price: { usd: srcTokenPrice },
+                    decimals,
+                },
+                destination_native_token: {
+                    token_price: { usd: destinationTokenPrice },
+                },
+                execute_gas_multiplier: multiplier = 1.1,
+            } = data.result;
+
+            const axelarBaseFee = Math.ceil(parseFloat(sourceBaseFee) * Math.pow(10, decimals));
+            const relativeGasPrice = Math.ceil(parseFloat(gasPrice) * parseFloat(multiplier));
+            const gasPriceRatio = parseFloat(destinationTokenPrice) / parseFloat(srcTokenPrice);
+            const relativeBlobBaseFee = Math.ceil(blobBaseFee * gasPriceRatio);
+
+            return [gasEstimationType, axelarBaseFee, relativeGasPrice, relativeBlobBaseFee];
+        }),
+    );
+}
+
+async function processCommand(config, chain, options) {
+    const {
+        env,
+        contractName,
+        address,
+        action,
+        privateKey,
+
+        chains,
+
+        destinationChain,
+        destinationAddress,
+        executionGasLimit,
+
+        yes,
+    } = options;
+
+    const contracts = chain.contracts;
+    const contractConfig = contracts[contractName];
+
+    let GasServiceAddress;
+
+    if (isValidAddress(address)) {
+        GasServiceAddress = address;
+    } else {
+        if (!contractConfig?.address) {
+            throw new Error(`Contract ${contractName} is not deployed on ${chain.name}`);
+        }
+
+        GasServiceAddress = contractConfig.address;
+    }
+
+    const rpc = chain.rpc;
+    const provider = getDefaultProvider(rpc);
+
+    const wallet = await getWallet(privateKey, provider, options);
+    await printWalletInfo(wallet, options);
+
+    printInfo('Contract name', contractName);
+    printInfo('Contract address', GasServiceAddress);
+
+    const gasService = new Contract(GasServiceAddress, getContractJSON('IAxelarGasService').abi, wallet);
+
+    const gasOptions = await getGasOptions(chain, options, contractName);
+
+    printInfo('GasService Action', action);
+
+    if (prompt(`Proceed with action ${action} on chain ${chain.name}?`, yes)) {
+        return;
+    }
+
+    switch (action) {
+        case 'estimateGasFee': {
+            validateParameters({
+                isNonEmptyString: { destinationChain },
+                isValidAddress: { destinationAddress },
+                isNumber: { executionGasLimit },
```

Comment:
Is this expected to be passed in as wei or eth? If eth then it needs to be converted to wei when calling the gas service right?

Reply:
It's in gas units